### PR TITLE
Run chromatic on branch ready

### PR DESCRIPTION
**Pull request by etienneburdet**
## Summary

The goal for this PR is to run chromatic only when the branch is ready is for review. The idea is that we would keep branches in draft state for longer and only turn them ready relatively late. It's always possible to discuss drafts or trigger manual runs.

For now:
- API client code coverage run for each commit where api-client files has been edited
- Viz CI run for each commit in a PR. Chromatic job run only when test and lint jobs passed with the PR state as 'ready for review'

Other possibilities include : run on label change, run on review request… for exemple.

**Review workflow**: 
- Create a new branch from this branch
- Add commits that should break the CI (test or lint)
- Open a PR in draft (CI should fail and chromatic job not run)
- Fix the CI, chromatic job should still not run
- Make the PR ready for review. CI should re-run with now chromatic running
- Make a commit that break lint or test. CI should fail, chromatic job should not run

### CODE REVIEW

**Comment by etienneburdet** [comment]
Ok this PR is officially "ready for review" (I had to turn it on and off to test stuff).  The conditional trigger of Chromatic on lint seems to work well.

I didn't manage to have the `if` working properly on this branch, for not running on draft. It seems it only works if the PR opened as draft, but if it has been turned on/off it still runs everything 🤷

File: .github/workflows/visualizations.yml [hunk 1]
```diff
@@ -0,0 +1,99 @@
...
+        run: npm ci
+
+      - name: Bootstrap
+        run: npm run bootstrap
+
+      - name: Lint
+        run: npm run lint
+
+  chromatic-deployment:
+    if: github.event.pull_request.draft == false
```
Comment (KevinFabre-ods) [inline .github/workflows/visualizations.yml:64]: Still run even if the PR in draft mode, see https://github.com/opendatasoft/ods-dataviz-sdk/pull/214

**Review by KevinFabre-ods — CHANGES_REQUESTED** [review]
Thanks for taking care of that:

Chromatic deployment: 
- is blocked when at least one test or lint job fails ✅ 
- is run only when the PR is review for review 🚫

Don't know why but the` if: github.event.pull_request.draft == false` condition doesn't block the job

**Comment by KevinFabre-ods** [comment]
In the viz workflow by replacing the 
```
on: 
 - push 
```
by
``` 
on: 
   pull_request:
      types:
        - opened
        - synchronize
        - reopened
        - ready_for_review
```

it works (https://github.com/opendatasoft/ods-dataviz-sdk/pull/214)

to access `github.event.pull_request.draft` we need to react to the pull_request event, that what we are doing by adding the on: pull_request

**Comment by etienneburdet** [comment]
Ha nice, I think I lacked the `synchronize` in the early version for it to react to push. I'll just rebase to erase test commits before merging.

**Comment by KevinFabre-ods** [comment]
I already rebased and erased tests commits => https://github.com/opendatasoft/ods-dataviz-sdk/pull/199/commits/73a3814cf26c7bb0cca03928a55ee1e4499f8a0b

**Review by KevinFabre-ods — APPROVED** [review]
Looks good to me now

@wmai you have review instructions in the PR description 😉

**Review by wmai — APPROVED** [review]
I've tried all sorts of choreographies, the CI behaves as expected, very good job, thank you for this work. 🙏

**Pull request merged**